tutorials/01_Basic_QA_Pipeline.ipynb

```diff
@@ -224,188 +154,149 @@
-    "\n",
-    "**Alternatives (Models):** e.g. \"distilbert-base-uncased-distilled-squad\" (fast) or \"deepset/bert-large-uncased-whole-word-masking-squad2\" (good accuracy)\n",
-    "\n",
-    "**Hint:** You can adjust the model to return \"no answer possible\" with the no_ans_boost. Higher values mean the model prefers \"no answer possible\"\n",
-    "\n",
-    "#### FARMReader"
+    "Initialize the `BM25Retriever`. For more Retriever options, see [Retriever](https://docs.haystack.deepset.ai/docs/retriever)"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {
-    "pycharm": {
-     "is_executing": false
-    }
-   },
+   "metadata": {},
    "outputs": [],
    "source": [
-    "from haystack.nodes import FARMReader\n",
-    "\n",
-    "# Load a  local model or any of the QA models on\n",
-    "# Hugging Face's model hub (https://huggingface.co/models)\n",
+    "from haystack.nodes import BM25Retriever\n",
     "\n",
-    "reader = FARMReader(model_name_or_path=\"deepset/roberta-base-squad2\", use_gpu=True)"
+    "retriever = BM25Retriever(document_store=document_store)"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "#### TransformersReader\n",
+    "## Initializing the Reader\n",
     "\n",
-    "Alternative:"
+    "Initialize the `FARMReader` with the `deepset/robert-base-squad2` model. For more Reader options, see [Reader](https://docs.haystack.deepset.ai/docs/reader)."
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {},
+   "metadata": {
+    "pycharm": {
+     "is_executing": false
+    }
+   },
    "outputs": [],
    "source": [
-    "from haystack.nodes import TransformersReader\n",
-    "# reader = TransformersReader(model_name_or_path=\"distilbert-base-uncased-distilled-squad\", tokenizer=\"distilbert-base-uncased\", use_gpu=-1)"
+    "from haystack.nodes import FARMReader\n",
+    "\n",
+    "reader = FARMReader(model_name_or_path=\"deepset/roberta-base-squad2\", use_gpu=True)"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "### Pipeline\n",
+    "## Creating the Retriever-Reader Pipeline\n",
     "\n",
-    "With a Haystack `Pipeline` you can stick together your building blocks to a search pipeline.\n",
-    "Under the hood, `Pipelines` are Directed Acyclic Graphs (DAGs) that you can easily customize for your own use cases.\n",
-    "To speed things up, Haystack also comes with a few predefined Pipelines. One of them is the `ExtractiveQAPipeline` that combines a retriever and a reader to answer our questions.\n",
-    "You can learn more about `Pipelines` in the [docs](https://haystack.deepset.ai/docs/latest/pipelinesmd)."
+    "The `ExtractiveQAPipeline` connects the Reader and Retriever. This makes the system fast because the Reader only processes the Documents that the Retriever has passed on."
```

Comment:
1. Related to the above: if I had no idea about Haystack and what the two nodes above are, I don't think I would make sense of this sentence.

2. One more long-standing question of mine is what's the value of `ExtractiveQAPipeline`. It goes a bit in the direction of `convert_files_to_docs` and actively hides one of the most powerful features of Haystack: the bare `Pipeline` object. Compare:

```python
from haystack.pipelines import ExtractiveQAPipeline

pipe = ExtractiveQAPipeline(reader, retriever)
```
with:
```python
from haystack import Pipeline

pipe = Pipeline()
pipe.add_node(retriever, name="retriever", inputs=["Query"])
pipe.add_node(reader, name="reader", inputs=["retriever"])
```

IMHO the latter says much more without adding too much complexity. It's easy to read. It's almost self-explanatory. You will have to explain `inputs=["Query"]`, that's true: but it gives room for users to experiment, change the components, rename them, swap them, add other nodes... We can even show them how to draw the pipeline with `pipe.draw('pipeline.graph.png')`. Isn't that nice? :grin: 

Maybe I'm too much of a power user at this point but I would not like to hide `Pipeline` to the users behind `ExtractiveQAPipeline`. It would also highlight the differences between indexing and query pipelines, if we replace `convert_files_to_docs`.

Reply:
I would agree and disagree here. I think `Pipeline`s are important and people should know more about their potential💯  
However, this is the first tutorial. The reader/user needs to learn about `document_store`, `reader` and `retriever` concepts, how to formulate a question with parameters etc. It is already a lot to digest. Therefore, I would advocate for adding a link to the `Creating the Retriever-Reader Pipeline` section, saying that "pipelines are customizable, refer here for more info".

Reply:
Fair point! It's true that we should not add too much info in the very first tutorial, so we should be careful what to add and where to stop. My point is more about how `ExtractiveQAPipeline` really helps. Does it simplify anything? Does it make the tutorial shorter? Not that much :smile: 

So why taking two lines to explain a toy class, when you can use four to introduce a core concept?

I'm not going to block anyone from going ahead with this version, but to me the `Pipeline` one still looks like a better idea :grin:

Reply:
Sounds good :blush: I was not considering the fact that Tut2 will not look like the current Tut2. With that in mind it should be fine to keep these little wrappers around.

Reply:
Yea for the first tutorial I would agree that the `ExtractiveQAPipeline` wrapper helps with the initial hurdle of getting started. 🙌